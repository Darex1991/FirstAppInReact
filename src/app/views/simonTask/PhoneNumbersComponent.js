import React, {Component, PropTypes} from 'react';
import {sortBy, clone} from "lodash";
import { Text } from 'react-form'

export default class PhoneNumbersComponent extends Component {
  constructor(props){
    super(props);
    this.state = {
      test: [{id:1, name: 'asd', key: 1, value: 'val 1'},{id:2, name: 'asdasdasd', key: 2, value: 'val 122222'},{id:3, name: 'asdasdasd', key: 3, value: 'val asdasd'},{id:4, name: '222', key: 4, value: 'val213123 1'}]
    }
  }

  goUp = (item, i) =>{
    const array = clone(this.state.test);
    array[i - 1].key += 1;
    array[i].key -= 1;
    const sortedArray = sortBy(array, 'key');
    this.setState({test: sortedArray});
    return false;
  };

  goDown = (item, i) =>{
    const array = clone(this.state.test);
    array[i + 1].key -= 1;
    array[i].key += 1;
    const sortedArray = sortBy(array, 'key');
    this.setState({test: sortedArray});
    return false;
  };

  render() {
      let phones = this.state.test.map( (item, i) => {
        return (
            <div className="col-xs-12" key={i} >
             <button disabled={item.key == 1} onClick={() =>this.goUp(item, i)}>GO UP</button>
             <button disabled={item.key == this.state.test.length} onClick={() =>this.goDown(item, i)}>GO DOWN</button>
             {item.name}
             <Text field={item.id}/>
          </div>
          )});

    return (
      <div className="phone default-input">
        <h2>Phone</h2>
        {phones}
      </div>
    )
  }
}
