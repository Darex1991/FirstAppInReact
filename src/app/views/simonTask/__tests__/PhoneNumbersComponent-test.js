import React from 'react';
import renderer from 'react-test-renderer';
import PhoneNumbersComponent from '../PhoneNumbersComponent';
test('Should render PhoneNumbersComponent correctly', () => {
    const tree = renderer.create(
        <PhoneNumbersComponent /> 
    ).toJSON();
    expect(tree).toMatchSnapshot();
});
    