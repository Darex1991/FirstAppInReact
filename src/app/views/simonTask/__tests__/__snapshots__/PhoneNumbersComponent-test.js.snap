exports[`test Should render PhoneNumbersComponent correctly 1`] = `
<div
  className="phone default-input">
  <h2>
     Phone 
  </h2>
  <AddNewPhoneComponent
    dataBase={
      Object {
        "phones": Array [
          Object {
            "checked": true,
            "highPriority": true,
            "id": 1,
            "key": 1,
            "name": "asd",
            "value": "val 1",
          },
          Object {
            "checked": false,
            "highPriority": false,
            "id": 2,
            "key": 2,
            "name": "asdasdasd",
            "value": "val 122222",
          },
          Object {
            "checked": false,
            "highPriority": true,
            "id": 3,
            "key": 3,
            "name": "asdasdasd",
            "value": "val asdasd",
          },
          Object {
            "checked": true,
            "highPriority": false,
            "id": 4,
            "key": 4,
            "name": "222",
            "value": "val213123 1",
          },
        ],
      }
    }
    updateDataBase={[Function]} />
  <div
    className="col-xs-12">
    <div
      className="col-xs-3">
      <button
        disabled={true}
        onClick={[Function]}>
         GO UP 
      </button>
      <button
        disabled={false}
        onClick={[Function]}>
         GO DOWN 
      </button>
    </div>
    <div
      className="col-xs-3">
      asd
    </div>
    <div
      className="col-xs-3">
      <InputWithoutValue
        onChange={[Function]}
        value="val 1" />
    </div>
  </div>
  <div
    className="col-xs-12">
    <div
      className="col-xs-3">
      <button
        disabled={false}
        onClick={[Function]}>
         GO UP 
      </button>
      <button
        disabled={false}
        onClick={[Function]}>
         GO DOWN 
      </button>
    </div>
    <div
      className="col-xs-3">
      asdasdasd
    </div>
    <div
      className="col-xs-3">
      <InputWithoutValue
        onChange={[Function]}
        value="val 122222" />
    </div>
  </div>
  <div
    className="col-xs-12">
    <div
      className="col-xs-3">
      <button
        disabled={false}
        onClick={[Function]}>
         GO UP 
      </button>
      <button
        disabled={false}
        onClick={[Function]}>
         GO DOWN 
      </button>
    </div>
    <div
      className="col-xs-3">
      asdasdasd
    </div>
    <div
      className="col-xs-3">
      <InputWithoutValue
        onChange={[Function]}
        value="val asdasd" />
    </div>
  </div>
  <div
    className="col-xs-12">
    <div
      className="col-xs-3">
      <button
        disabled={false}
        onClick={[Function]}>
         GO UP 
      </button>
      <button
        disabled={true}
        onClick={[Function]}>
         GO DOWN 
      </button>
    </div>
    <div
      className="col-xs-3">
      222
    </div>
    <div
      className="col-xs-3">
      <InputWithoutValue
        onChange={[Function]}
        value="val213123 1" />
    </div>
  </div>
</div>
`;
